library/core/src/array/mod.rs

```diff
@@ -364,3 +364,30 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[lang = "array"]
+#[cfg(not(bootstrap))]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same aize as self, with `f` applied to each element.
+    ///
+    /// # Examples
+    /// ```
+    /// let x = [1,2,3];
+    /// let y = x.map(|v| v + 1);
+    /// assert_eq!(y, [2,3,4]);
+    /// ```
+    #[unstable(feature = "array_map", issue = "77777")]
+    fn map<F, S>(self, f: F) -> [S; N]
+    where
+        F: FnMut(T) -> S,
+    {
+        use crate::mem::MaybeUninit;
+        let dst = MaybeUninit::uninit_array::<N>();
+        for (i, e) in self.into_iter().enumerate() {
+            dst[i] = MaybeUninit::new(f(e));
```

Comment:
This looks like it would leak all previous elements if f panics

Reply:
Do you mean of `dst` or of the iterator?

Reply:
of `dst` (since `MaybeUninit` won't run the destructor of the contained value)

Reply:
Of dest because it's MaybeUninit so nothing will get dropped automatically.

Reply:
`dst`, see the implementation of `fn make` on how this can be handled:
https://github.com/rust-lang/rust/blob/c448b373b57d0f1adefb70e07a44b1cccc9d2393/library/core/src/array/mod.rs#L357-L385